doc/pandas-2.0/source/goals.rst

```diff
@@ -0,0 +1,194 @@
+  very obvious accumulations of technical debt.
+* pandas is being used increasingly for very large datasets on machines with
+  many cores and large amounts of RAM (100s of gigabytes to terabytes). It
+  would be nice to be able to better utilize these larger, beefier systems
+  within a single Python process.
+* pandas is being used increasingly as a computational building block of some
+  larger system, such as Dask or Apache Spark. We should consider reducing the
+  overhead for making data accessible to pandas (i.e. via memory-mapping or
+  other low-overhead memory sharing).
+* Rough edges in pandas's implementation (e.g. its handling of missing data
+  across data types) are being exposed to users.
+
+These documents are largely concerned with pandas's internal design, which is
+mostly invisible to average users. Advanced users of pandas are generally
+familiar with some of these internal details, particular around performance and
+memory use, and so the degree to which users are impacted will vary quite a
+lot.
+
+Key areas of work
+=================
+
+Possible changes or improvements to pandas's internals fall into a number of
+different buckets to be explored in great detail:
+
+* **Decoupling from NumPy while preserving interoperability**: by eliminating
+  the presumption that pandas objects internally must contain data stored in
+  NumPy ``ndarray`` objects, we will be able to bring more consistency to
+  pandas's semantics and enable the core developers to extend pandas more
+  cleanly with new data types, data structures, and computational semantics.
+* **Exposing a pandas Cython and/or C/C++ API to other Python library
+  developers**: the internals of Series and DataFrame are only weakly
+  accessible in other developers' native code. At minimum, we wish to better
+  enable developers to construct the precise data structures / memory
+  representation that fill the insides of Series and DataFrame.
+* **Improving user control and visibility of memory use**: pandas's memory use,
+  as a result of its internal implementation, can frequently be opaque to the
+  user or outright unpredictable.
+* **Improving performance and system utilization**: We aim to improve both the
+  micro (operations that take < 1 ms) and macro (all other operations)
+  performance of pandas across the board. As part of this, we aim to make it
+  easier for pandas's core developers to leverage multicore systems to
+  accelerate computations (without running into any of Python's well-known
+  concurrency limitations)
+* **Removal of deprecated / underutilized functionality**: As the Python data
+  ecosystem has grown, a number of areas of pandas (e.g. plotting and datasets
+  with more than 2 dimensions) may be better served by other open source
+  projects. Also, functionality that has been explicitly deprecated or
+  discouraged from use (like the ``.ix`` indexing operator) would ideally be
+  removed.
```

Comment:
Yes, I would either move sparse into libpandas (if it can be done cleanly) or drop it. But I would hesitate to hold up the rest of pandas 2.0 for the sake of sparse, which IMO is pretty niche.